crates/oxc_linter/src/rules/eslint/sort_keys.rs

```diff
@@ -0,0 +1,1089 @@
+                        property_groups.push(vec!["<linebreak_group>".into()]);
+                        property_groups.push(vec![]);
+                        continue;
+                    }
+                }
+                property_groups.last_mut().unwrap().push(key.into());
+            }
+
+            if !self.case_sensitive {
+                for group in &mut property_groups {
+                    *group = group.iter().map(|s| s.to_lowercase()).collect::<Vec<String>>();
+                }
+            }
+
+            let mut sorted_property_groups = property_groups.clone();
+            for group in &mut sorted_property_groups {
+                if self.natural {
+                    natural_sort(group);
+                } else {
+                    alphanumeric_sort(group);
+                }
+
+                if self.sort_order == SortOrder::Desc {
+                    group.reverse();
+                }
+            }
+
+            let is_sorted =
+                all(property_groups.iter().zip(&sorted_property_groups), |(a, b)| a == b);
+
+            if !is_sorted {
+                ctx.diagnostic(sort_properties_diagnostic(node.span()));
+            }
+        }
+    }
+}
+
+fn alphanumeric_cmp(a: &str, b: &str) -> Ordering {
+    /* regex key special case */
+    if a.starts_with('/') && a.ends_with('/') {
+        if b.starts_with('/') && b.ends_with('/') {
+            return a.cmp(b);
+        }
+        return Ordering::Greater;
+    }
+
+    if b.starts_with('/') && b.ends_with('/') {
+        return Ordering::Less;
+    }
+
+    /* empty keys special case */
+    if a.is_empty() && b.starts_with('[') || b.is_empty() && a.starts_with('[') {
+        return Ordering::Equal;
+    }
+
+    let len = a.len().min(b.len());
+    let a_chars: Vec<char> = a.chars().take(len).collect();
+    let b_chars: Vec<char> = b.chars().take(len).collect();
+
+    for (a_char, b_char) in a_chars.iter().zip(b_chars.iter()) {
+        if a_char == b_char {
+            continue;
+        }
+        /* JS sorting apparently places _ after uppercase alphanumerics and before lowercase ones */
+        if a_char.is_uppercase() && *b_char == '_' {
+            return Ordering::Less;
+        }
+
+        if *a_char == '_' && b_char.is_uppercase() {
+            return Ordering::Greater;
+        }
+
+        /* computed properties should come before alpha numeric chars */
+        if *a_char == '[' && b_char.is_alphanumeric() {
+            return Ordering::Less;
+        }
+
+        if a_char.is_alphanumeric() && *b_char == '[' {
+            return Ordering::Greater;
+        }
+
+        if a_char.is_alphanumeric() && !b_char.is_alphanumeric() {
+            return Ordering::Greater;
+        }
+
+        if !a_char.is_alphanumeric() && b_char.is_alphanumeric() {
+            return Ordering::Less;
+        }
+
+        return a_char.cmp(b_char);
+    }
+
+    a.cmp(b)
```

Comment:
Can we not return `Ordering::Equal` here?

Reply:
I am not 100% sure, thats why I kept it like this.